modules/sdk/gradle-plugins-workspace/src/main/java/com/liferay/gradle/plugins/workspace/configurator/RootProjectConfigurator.java

```diff
@@ -1863,6 +1869,67 @@ public void execute(
 			});
 	}
 
+	private void _configurePropertiesTask(Project project) {
+		project.afterEvaluate(
+			new Action<Project>() {
+
+				@Override
+				public void execute(Project project) {
+					TaskContainer taskContainer = project.getTasks();
+
+					Task propertiesTask = taskContainer.findByName(
+						"properties");
+
+					if (Objects.isNull(propertiesTask)) {
+						return;
+					}
+
+					propertiesTask.doFirst(
+						new Action<Task>() {
+
+							@Override
+							public void execute(Task task) {
+								File propertiesFile = project.file(
+									"gradle-local.properties");
+
+								if (!propertiesFile.exists()) {
+									return;
+								}
+
+								Properties properties = new Properties();
+
+								try (BufferedReader bufferedReader =
+										Files.newBufferedReader(
+											propertiesFile.toPath())) {
+
+									properties.load(bufferedReader);
+								}
+								catch (Exception exception) {
+									throw new GradleException(
+										"Unable to read gradle-local." +
+											"properties",
+										exception);
+								}
+
+								ExtensionContainer projectExtensions =
+									project.getExtensions();
+
+								ExtraPropertiesExtension
+									extraPropertiesExtension =
+										projectExtensions.getExtraProperties();
+
+								for (Object key : properties.keySet()) {
+									extraPropertiesExtension.set(
+										key.toString(), properties.get(key));
+								}
+							}
+
+						});
+				}
+
+			});
+	}
+
```

Comment:
@simonjhy the purpose of the plugin is to read those properties, we should not have to do this manually. We need to look into why the properties task is not reading the override properties found in the `gradle-local.properties` file.

My guess is that the plugin needs to be applied to projects, not just the settings.

Reply:
OK. got it.